homeassistant/helpers/config_entry_oauth2_flow.py

```diff
@@ -259,10 +259,21 @@ def extra_authorize_data(self) -> dict:
         """
         return self.async_create_entry(title=self.flow_impl.name, data=data)
 
-    async_step_user = async_step_pick_implementation
-    async_step_ssdp = async_step_pick_implementation
-    async_step_zeroconf = async_step_pick_implementation
-    async_step_homekit = async_step_pick_implementation
+    async def async_step_discovery(self, user_input: dict = None) -> dict:
+        """Handle a flow initialized by discovery."""
+        await self.async_set_unique_id(self.DOMAIN)
+        self._abort_if_unique_id_configured()
+
+        assert self.hass is not None
+        if self.hass.config_entries.async_entries(self.DOMAIN):
```

Comment:
This is not necessary because of unique id abort

Reply:
So if a user does a manual flow, or has the integration configured already, we should keep discovering it?

I think it is better to stop "recommending" (since that is basically what it is).

Reply:
Let's just always set unique ID, even for user flows.

Reply:
Uh? So in the case of Spotify for example, how would that work with multiple accounts?

Reply:
You are right. That won't work. So let's not do it.